sound/hda/ext/hdac_ext_stream.c

```diff
@@ -310,13 +311,41 @@ hdac_ext_host_stream_assign(struct hdac_bus *bus,
 		spin_lock_irq(&bus->reg_lock);
 		res->hstream.opened = 1;
 		res->hstream.running = 0;
-		res->hstream.substream = substream;
+		res->hstream.substream = NULL;
+		res->hstream.cstream = NULL;
```

Comment:
@crojewsk is this really necessary to initialize these to NULL here?

Reply:
Yeah as pcm/ compr null-check these fields. In current state, pcm (substream) is always tied to hdac_stream after assignment. After compr-support addition that is no longer true - there is always pcm (substream) OR compr (cstream) tied to hdac_stream after assignment.

Reply:
I see, thanks!